### ISSUE TITLE
#286 breaks Emacs 26.3

### ISSUE THREAD

**Issue by scop** [editorconfig/editorconfig-emacs#289]
#286 breaks this mode with Emacs 26.3 for me. Reverting to b8043702f3d977db0e030c6c64ee4a810cad5f45 fixes it.

Warnings:
```
Warning (editorconfig): Failed to get properties, styles will not be applied:
(editorconfig-error "Error from editorconfig-get-properties-function:
    (invalid-function (val (gethash key result)))")
```
----

straight.el byte compilation output as of ce5d4358394252f506fc789d46368621dbaaf064:
```
$ /usr/bin/emacs -Q -L .../.emacs.d/straight/build/editorconfig/ --batch --eval \
 "(let ((default-directory \".../.emacs.d/straight/build/\"))
  (normal-top-level-add-subdirs-to-load-path)
  (byte-recompile-directory \".../.emacs.d/straight/build/editorconfig/\" 0 'force))"
Checking .../.emacs.d/straight/build/editorconfig/...
Compiling .../.emacs.d/straight/build/editorconfig/editorconfig-autoloads.el...
Compiling .../.emacs.d/straight/build/editorconfig/editorconfig-conf-mode.el...
Compiling .../.emacs.d/straight/build/editorconfig/editorconfig-core-handle.el...
Compiling .../.emacs.d/straight/build/editorconfig/editorconfig-core.el...

In toplevel form:
editorconfig-core.el:91:1:Warning: Unused lexical argument ‘directory’

In editorconfig-core-get-nearest-editorconfig:
editorconfig-core.el:91:52:Warning: ‘(handle (car (last
    (editorconfig-core--get-handles directory ".editorconfig"))))’ is a
    malformed function
editorconfig-core.el:93:15:Warning: reference to free variable ‘handle’

In editorconfig-core-get-properties-hash:
editorconfig-core.el:123:71:Warning: ‘(val (gethash key result))’ is a
    malformed function
editorconfig-core.el:148:32:Warning: reference to free variable ‘val’

In end of data:
editorconfig-core.el:180:1:Warning: the function ‘when-let’ is not known to be
    defined.
Compiling .../.emacs.d/straight/build/editorconfig/editorconfig-fnmatch.el...
Compiling .../.emacs.d/straight/build/editorconfig/editorconfig.el...

In editorconfig-set-indentation-python-mode:
editorconfig.el:403:36:Warning: assignment to free variable
    ‘python-indent-offset’

In editorconfig-find-current-editorconfig:
editorconfig.el:894:4:Warning: ‘(file
    (editorconfig-core-get-nearest-editorconfig default-directory))’ is a
    malformed function
editorconfig.el:898:16:Warning: reference to free variable ‘file’

In end of data:
editorconfig.el:974:1:Warning: the function ‘when-let’ is not known to be
    defined.
Done (Total of 5 files compiled, 1 skipped)
```

**Comment by jcs090218** [comment]
I'm on it!

**Comment by jcs090218** [comment]
> #286 breaks this mode

Can you elaborate what's not working? Does it break while installing it? Editing files?, etc.

Here is the result using Eask:

```
$> eask install
Running Eask in the development environment
Press Ctrl+C to cancel.

Executing script inside Emacs...

  Checking Emacs version 26.3... done!
  Checking system windows-nt... done!
  Try constructing the package-descriptor (editorconfig.el)... succeeded!
Unmatched keyword 'convenience'; add ;; Keywords ... convenience to package-file or consider removing it
Unmatched keyword 'editorconfig'; add ;; Keywords ... editorconfig to package-file or consider removing it
  Loading Eask file in c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/Eask... done!

Loading package information... done
Installing 1 package dependency...
  - Skipping nadvice (0.4)... already installed
(Total of 0 dependency installed, 1 skipped)
Searching for artifact to install...
  Found artifact in c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/dist/editorconfig-0.9.1.tar
Parsing tar file...
Parsing tar file...done
Parsing tar file...
Parsing tar file...done
Extracting editorconfig-0.9.1/
Extracting editorconfig-0.9.1/editorconfig-conf-mode.el
Extracting editorconfig-0.9.1/editorconfig-core-handle.el
Extracting editorconfig-0.9.1/editorconfig-core.el
Extracting editorconfig-0.9.1/editorconfig-fnmatch.el
Extracting editorconfig-0.9.1/editorconfig-pkg.el
Extracting editorconfig-0.9.1/editorconfig.el
Generating autoloads for editorconfig-conf-mode.el...
Generating autoloads for editorconfig-conf-mode.el...done
Generating autoloads for editorconfig-core-handle.el...
Generating autoloads for editorconfig-core-handle.el...done
Generating autoloads for editorconfig-core.el...
Generating autoloads for editorconfig-core.el...done
Generating autoloads for editorconfig-fnmatch.el...
Generating autoloads for editorconfig-fnmatch.el...done
Generating autoloads for editorconfig-pkg.el...
Generating autoloads for editorconfig-pkg.el...done
Generating autoloads for editorconfig.el...
Generating autoloads for editorconfig.el...done
Wrote c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/.eask/26.3/elpa/editorconfig-0.9.1/editorconfig-autoloads.el
Wrote c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/.eask/26.3/elpa/editorconfig-0.9.1/editorconfig-autoloads.el
Checking c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/.eask/26.3/elpa/editorconfig-0.9.1...
Compiling c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/.eask/26.3/elpa/editorconfig-0.9.1/editorconfig-autoloads.el...
Compiling c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/.eask/26.3/elpa/editorconfig-0.9.1/editorconfig-conf-mode.el...
Compiling c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/.eask/26.3/elpa/editorconfig-0.9.1/editorconfig-core-handle.el...
Compiling c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/.eask/26.3/elpa/editorconfig-0.9.1/editorconfig-core.el...
Compiling c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/.eask/26.3/elpa/editorconfig-0.9.1/editorconfig-fnmatch.el...
Compiling c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/.eask/26.3/elpa/editorconfig-0.9.1/editorconfig-pkg.el...
Compiling c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/.eask/26.3/elpa/editorconfig-0.9.1/editorconfig.el...

In editorconfig-set-indentation-python-mode:
editorconfig.el:403:36:Warning: assignment to free variable
    `python-indent-offset'
Done (Total of 5 files compiled, 2 skipped)

(Installed in c:/Users/JenChieh/Downloads/workspace/editorconfig-emacs/.eask/26.3/elpa/editorconfig-0.9.1/)
```

I don't use straight, would it be an issue from them? 😕

**Comment by scop** [comment]
Breaks as in shows the warnings from the initial comment on loading a file, `.editorconfig` settings are not applied.

I see the above output operates on a 0.9.1 tarball. Is that the actual 0.9.1 release tarball, or something generated from current master? My issue is with current master.

**Comment by jcs090218** [comment]
Yeah, the output is from the master branch, not the 0.9.1 tarball. I have tried another test but still unable to reproduce the error. My best guess is to reinstall the package cleanly? Let me know if you found something strange. 😓

**Comment by scop** [comment]
Moved my `~/.emacs.d` and `~/.emacs` out of the way, followed https://github.com/editorconfig/editorconfig-emacs#manual-installation -> same problem.

**Comment by cweiske** [comment]
Could you please tag that as 0.10.1, so that it is made available in melpa?
Manually installing this package is harder than I thought.

**Comment by scop** [comment]
With current master, the mode is no longer completely broken, but the same warning as in the initial comment still pops up on load.

> Warning (editorconfig): Failed to get properties, styles will not be applied: (editorconfig-error "Error from editorconfig-get-properties-function: (invalid-function (val (gethash key result)))")

**Comment by groengpx** [comment]
I have identical symptoms.

**Comment by 10sr** [comment]
> Could you please tag that as 0.10.1, so that it is made available in melpa?

The commit is already included in 0.10.0. It seems there still be another issue...

**Comment by jcs090218** [comment]
Would the following patch work? 🤔 

```diff
diff --git a/editorconfig-core.el b/editorconfig-core.el
index d5f46fe..f6f8f84 100644
--- a/editorconfig-core.el
+++ b/editorconfig-core.el
@@ -144,8 +144,9 @@ hash object instead."
     ;; Downcase known boolean values
     (dolist (key '( end_of_line indent_style indent_size insert_final_newline
                     trim_trailing_whitespace charset))
-      (when-let ((val (gethash key result)))
-        (puthash key (downcase val) result)))
+      (let ((val (gethash key result)))
+        (when val
+          (puthash key (downcase val) result))))

     ;; Add indent_size property
     (let ((v-indent-size (gethash 'indent_size result))
```

**Comment by 10sr** [comment]
That should be ok too 👍 , but I googled and found when-let* might be another solution.

**Comment by 10sr** [comment]
Released v0.10.1, please try this version 🙏

**Comment by groengpx** [comment]
Thanks, it seems to work now, without warnings.
